tiny_cnn/layers/convolutional_layer.h

```diff
@@ -406,7 +310,31 @@ class convolutional_layer : public feedforward_layer<Activation> {
         init();
     }
 
-private:
+ private:
+    void init_backend(backend_t backend_type) {
+        using namespace std::placeholders;  // for _1, _2, _3...
+        switch (backend_type) {
+            case backend_t::tiny_cnn:
+                Base::backend_ = std::make_shared<core::tiny_backend>(&params_,
+                    std::bind(&convolutional_layer::copy_and_pad_input, this, _1, _2),
+                    std::bind(&convolutional_layer::copy_and_unpad_delta, this, _1, _2),
+                    std::bind(&Base::backward_activation, this, _1, _2, _3),
+                    &conv_layer_worker_storage_);
+                Base::backend_->set_layer(this);
```

Comment:
@patrikhuber I'm not used with lambdas.
After a first search I would go for the following approach:
   
       Base::backend_ = std::make_shared<core::tiny_backend>(
           &params_,
           static_cast<std::function<void(const vec_t&, int)>>(copy_and_pad_input),
           static_cast<std::function<void(const vec_t&, vec_t&)>>(copy_and_unpad_delta),
           static_cast<std::function<void(const vec_t&, const vec_t&, vec_t&)>> (backward_activation),
           &conv_layer_worker_storage_);

where

      auto copy_and_pad_input   = [](const vec_t& in, int worker_index) { }
      auto copy_and_unpad_delta = [](const vec_t& delta, vec_t& dst) { }